### PR TITLE
Attempt to improve efficiency of read_parquet by using ChainedVector …

### PR DESCRIPTION
**Pull request by quinnj**
…for multichunk tables

cc: @tanmaykm @xiaodaigh 

This proposes a dedicated `Parquet.Table` type to be returned from `read_parquet`. It should be non-breaking, since you can "use" the result the same as you did before; i.e. `DataFrame(read_parquet(file))` still works. This avoids the call to `reduce(vcat, ...)` which can be quite costly for very large files with many chunks. It instead proposes to lazily "vcat" chunks using the `SentinelArrays.ChainedVector` array type. This is the same approach used by CSV.jl, Arrow.jl, and the general arrow c++/pyarrow projects (though they call their array type `ChunkedArray`).

This is my first foray into the parquet code, so forgive if I've assumed something wrongly or misstepped. I'm not quite sure on how the testing is setup, but I wanted to put up the PR to get review first and see if anything breaks unintentionally.

### CODE REVIEW

**Comment by xiaodaigh** [comment]
Great idea! It didn't occur to me that Julia ecosystem should define their own types like this but it's the logical approach to not rely on others.

File: src/simple_reader.jl [hunk 1]
```diff
@@ -16,16 +33,8 @@ function read_parquet(path)
...
-    for key in column_names
-        # combine all chunks' key into one column
-        one_column = reduce(vcat, chunk[key] for chunk in chunks)
-        result_dict[key] = one_column
-    end
-
-    result_dict
+    sch = Tables.schema(chunks[1])
+    N = length(sch.names)
+    columns = length(chunks) == 1 ? AbstractVector[chunks[1][i] for i = 1:N] : AbstractVector[ChainedVector([chunks[j][i] for j = 1:length(chunks)]) for i = 1:N]
```
Comment (xiaodaigh) [inline src/simple_reader.jl:38]:
```suggestion
    sch = Tables.schema(chunks[1])
    ncols = length(sch.names)
    columns = length(chunks) == 1 ? AbstractVector[chunks[1][i] for i = 1:ncols] : AbstractVector[ChainedVector([chunks[j][i] for j = 1:length(chunks)]) for i = 1:ncols]
```

**Review by xiaodaigh — APPROVED** [review]

**Comment by tanmaykm** [comment]
:+1: LGTM! Will add a commit in a bit to make the tests pass.

**Comment by tanmaykm** [comment]
https://github.com/JuliaIO/Parquet.jl/pull/130 should make the tests pass

**Comment by tanmaykm** [comment]
With some changes, it seems possible to load the chunks lazily on demand as well. That may be a good future enhancement.

**Comment by tanmaykm** [comment]
CI passes, will merge this in a bit

**Pull request merged**